contracts/token/ZNSSubdomainToken.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { ARegistryWired } from "../registry/ARegistryWired.sol";
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import { ERC721Enumerable } from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import { ERC721Votes } from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Votes.sol";
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import { ERC721URIStorage } from "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import { ERC2981 } from "@openzeppelin/contracts/token/common/ERC2981.sol";
+
+
+contract ZNSSubdomainToken is
+    ERC721,
+    ERC721Enumerable,
+    ERC721URIStorage,
+    EIP712,
+    ERC721Votes,
+    ERC2981,
+    ARegistryWired {
+
+    uint256 private _totalSupply;
+
+    string private baseURI;
+
+    event TokenURISet(uint256 indexed tokenId, string tokenURI);
+
+    event BaseURISet(string baseURI);
+
+    event DefaultRoyaltySet(uint96 indexed defaultRoyalty);
+
+    event TokenRoyaltySet(uint256 indexed tokenId, uint96 indexed royalty);
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        string memory version_,
+        address registry_,
+        address accessController_
+    )
+        ERC721(name_, symbol_)
+        EIP712(name_, version_)
+    {
+        _setRegistry(registry_);
+    }
+
+    function register(address to, uint256 tokenId, string memory _tokenURI)
+        external
+    {
+        ++_totalSupply;
+        _safeMint(to, tokenId);
+        _setTokenURI(tokenId, _tokenURI);
+    }
+
+    function revoke(uint256 tokenId)
+        external
+    {
+        _burn(tokenId);
```

Comment:
Be sure access control is on this function as well, when it is ready to be implemented.